ria_remote/create_sibling_ria.py

```diff
@@ -0,0 +1,305 @@
+        disabled_hook = sh_quote(str(repo_path / 'hooks' / 'post-update.sample'))
+        enabled_hook = sh_quote(str(repo_path / 'hooks' / 'post-update'))
+
+        if ssh_host:
+            from datalad import ssh_manager
+            ssh = ssh_manager.get_connection(ssh_host, use_remote_annex_bundle=False)
+            ssh.open()
+            ssh('cd {} && git init --bare'.format(sh_quote(str(repo_path))))
+            if not no_server:
+                ssh('mv {} {}'.format(disabled_hook, enabled_hook))
+        else:
+            cmd = ['git', 'init', '--bare']
+            subprocess.run(cmd, cwd=str(repo_path), check=True)
+            if not no_server:
+                cmd = ['mv', disabled_hook, enabled_hook]
+                subprocess.run(cmd, cwd=str(repo_path), check=True)
+
+        # add a git remote to the bare repository
+        # Note: needs annex-ignore! Otherwise we might push into default annex/object tree instead of
+        # directory type tree with dirhash lower. This in turn would be an issue, if we want to pack the entire thing
+        # into an archive. Special remote will then not be able to access content in the "wrong" place within the
+        # archive
+        lgr.debug("set up git remote")
+        # TODO: - This results in "[WARNING] Failed to determine if datastore carries annex."
+        #         (see https://github.com/datalad/datalad/issues/4028)
+        #       - additionally there's https://github.com/datalad/datalad/issues/3989, where datalad-siblings might
+        #         hang forever
+
+        if ssh_host:
+            lgr.error("XXXXXXXXXXXX\n%s", [str(p) for p in ssh_manager.socket_dir.iterdir()])
+            lgr.error("XXXXXXXXXXXX\n%s %s", ssh.ctrl_path, ssh.is_open())
+        ds.siblings(
+            'configure',
+            name=sibling,
+            url='{}:{}'.format(ssh_host, str(repo_path))
+            if ssh_host
+            else str(repo_path),
+            recursive=False,
+            publish_depends=storage_sibling,
+            result_renderer=None,
+            fetch=False)
+
+        ds.config.set("remote.{}.annex-ignore".format(sibling), value="true", where="local")
+        yield get_status_dict(
+            status='ok',
+            **res_kwargs,
+        )
+
+        if not no_publish:
+            # Publish to the git remote (without data)
+            # This should prevent weird disconnected history situations
+            # and give the remote end an idea who's dataset that is
+            lgr.info("updating sibling {}".format(sibling))
+            yield from ds.publish(to=sibling, transfer_data='none')
+
+        if recursive:
+            # Note: subdatasets can be treated independently, so go full recursion when querying for them and _no_
+            # recursion with the actual call. Theoretically this can be parallelized.
+            for subds in ds.subdatasets(fulfilled=True, recursive=True, result_xfm='datasets'):
```

Comment:
For consistency we should support `--recursion-limit` and pass it to `subdatasets()`

Reply:
Done